pkg/threescale/backend/manager.go

```diff
@@ -603,3 +610,20 @@ func getPeriodForLimit(ascendingPeriods []api.Period, limits map[api.Period]*Lim
 	}
 	return period, nil, false
 }
+
+func getMetricsFromApplicationForPeriod(app *Application, period api.Period) api.Metrics {
+	metrics := make(api.Metrics)
+
+	for metric, value := range app.UnlimitedCounter {
+		metrics.Add(metric, value)
+	}
+
+	for metric, value := range app.LimitCounter {
+		for timePeriod, counter := range value {
+			if timePeriod == period {
+				metrics.Add(metric, counter.current)
+			}
+		}
```

Comment:
rather than iterating, can't we just try to access `value[period]` and see if there's an entry for it since it is a map?

Reply:
maybe, this isn't used as of now so ignoring until i decide if we need it or not then ill either update or delete the code entirely :)